provider/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -204,44 +205,74 @@ fun kotlinCoreEnvironmentFor(configuration: CompilerConfiguration, rootDisposabl
 
 
 internal
-fun messageCollectorFor(log: Logger, pathTranslation: (String) -> String = { it }): MessageCollector =
+fun messageCollectorFor(log: Logger, pathTranslation: (String) -> String = { it }): LoggingMessageCollector =
+    LoggingMessageCollector(log, pathTranslation)
+
 
-    object : MessageCollector {
+internal
+data class ScriptCompilationError(val message: String, val location: CompilerMessageLocation?)
 
-        var errors = 0
 
-        override fun hasErrors() = errors > 0
+internal
+data class ScriptCompilationException(val errors: List<ScriptCompilationError>) : RuntimeException() {
+    override val message: String?
+        get() {
+            val errorPlural = if (errors.size > 1) "errors" else "error"
+            val maxLineNumberLen = errors.mapNotNull { it.location?.line }.max().toString().length
+            return (listOf("Script compilation $errorPlural:") + errors.map { error ->
+                if (error.location != null) {
+                    "${error.location.line.toString().padStart(maxLineNumberLen, '0')}: ${error.location.lineContent}\n" +
+                        "${" ".repeat(maxLineNumberLen + 1 + error.location.column)}^- ${error.message}"
```

Comment:
I wish the "up" part of this arrow character was more pronounced, it would work perfectly here
⬑
Other potential alternatives would be these:
 - ⇪
 - ▲
 - ⇧

This is the site I'm getting these examples from:
http://xahlee.info/comp/unicode_arrows.html

This works fine as is, but I wish that there was a character that would fit this use case better.